feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-page-layout-handler.ts

```diff
@@ -12,72 +12,125 @@ import {
+  protected static sectionNavigationDisplayOnly = 'navigationDisplayOnly';
+  protected static sectionHeader = 'header';
+  protected static sectionNavigation = 'navigation';
   constructor(
     protected configuratorRouterExtractorService: ConfiguratorRouterExtractorService,
     protected breakpointService: BreakpointService,
     protected layoutConfig: LayoutConfig,
     protected commonConfiguratorUtilsService: CommonConfiguratorUtilsService
   ) {}
+
   handle(
     slots$: Observable<string[]>,
     pageTemplate?: string,
     section?: string
   ) {
     if (
       pageTemplate === CpqConfiguratorPageLayoutHandler.templateName &&
-      section === 'header'
+      section === CpqConfiguratorPageLayoutHandler.sectionHeader
     ) {
-      this.configuratorRouterExtractorService
-        .extractRouterData()
-        .pipe(
-          switchMap((routerData) =>
-            this.breakpointService.isUp(BREAKPOINT.lg).pipe(
-              map((isLargeResolution) => ({
-                isLargeResolution: isLargeResolution,
-                routerData,
-              }))
-            )
-          ),
-          take(1)
-        )
+      this.compileRouterAndResolution()
+        .pipe(take(1))
+        .subscribe((cont) => {
+          slots$ = slots$.pipe(
+            map((slots) => {
+              return this.getHeaderSlots(slots, cont);
+            })
+          );
+        });
+    } else if (
+      pageTemplate === CpqConfiguratorPageLayoutHandler.templateName &&
+      section === CpqConfiguratorPageLayoutHandler.sectionNavigation
+    ) {
+      this.compileRouterAndResolution()
+        .pipe(take(1))
         .subscribe((cont) => {
           slots$ = slots$.pipe(
             map((slots) => {
-              if (
-                cont.routerData.pageType ===
-                ConfiguratorRouter.PageType.CONFIGURATION
-              ) {
-                const extendedSlots = ['PreHeader'];
-                extendedSlots.push(...slots);
-                return extendedSlots;
-              } else if (cont.routerData.displayOnly) {
-                if (cont.isLargeResolution) {
-                  return this.commonConfiguratorUtilsService.getSlotsFromLayoutConfiguration(
-                    this.layoutConfig,
-                    CpqConfiguratorPageLayoutHandler.templateName,
-                    CpqConfiguratorPageLayoutHandler.sectionDisplayOnlyName,
-                    BREAKPOINT.lg
-                  );
-                } else {
-                  return this.commonConfiguratorUtilsService.getSlotsFromLayoutConfiguration(
-                    this.layoutConfig,
-                    CpqConfiguratorPageLayoutHandler.templateName,
-                    CpqConfiguratorPageLayoutHandler.sectionDisplayOnlyName,
-                    BREAKPOINT.xs
-                  );
-                }
-              } else return slots;
+              return this.getNavigationSlots(slots, cont);
```

Comment:
You could just do 
`map((slots) => this.getNavigationSlots(slots, cont)`
Instead of {return this.getNavigationSlots(slots, cont);}